### ISSUE TITLE
Support for Boltzmann machines and Deep Boltzmann

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#770]
https://en.wikipedia.org/wiki/Boltzmann_machine
https://www.youtube.com/watch?v=p4Vh_zMw-HQ

**Comment by khalibartan** [comment]
@ankurankan  I can work on this. Do you have anything specific in your mind ?

**Comment by ankurankan** [comment]
@khalibartan No, just the normal Boltzmann machine with fit and predict methods. Just keep in mind that you might have to implement a few things in Undirected models to implement this.

**Comment by kris-singh** [comment]
@khalibartan @ankurankan  i would love to write the contrastive divergence code required to train the rbm.
I think we need to define this as follows we can have method adding_layer(self,hidden,visible) in the undirected model. The initialization of the weight matrix has to zeros i think and since both layers are always fully connected(i might be wrong assuming this). is this line of thinking correct??

**Comment by ankurankan** [comment]
@kris-singh I would suggest you not to work on this as I already have the code for this (I recently wrote Boltzmann Machine for a course) so I will just modify it and push it.

**Comment by kris-singh** [comment]
@ankurankan  great................have you implemented the CD algorithm aslo

**Comment by ankurankan** [comment]
@kris-singh No. Just a simple Boltzmann Machine.

**Comment by kris-singh** [comment]
@ankurankan then what about the inference from the model.

**Comment by ankurankan** [comment]
@kris-singh I have implementation for inference.
